Megaparsec/Megaparsec.lean

```diff
@@ -64,26 +64,26 @@ class Stream (S : Type) where
 -- Convention for optimising currying is to apply E, S then M, and leave A for the last type argument.
 -- Except for Result, Reply and ParseError where the order is S E A.
 
-inductive ErrorFancy (E: Type) where
+inductive ErrorFancy where
 | fail (msg: String)
 | indent (ord: Ordering) (fromPos: Pos) (uptoPos: Pos)
 | custom (e: E)
 
-inductive ParseError (E: Type) [Stream S] where
-| trivial (offset: Nat) (unexpected: ErrorItem (Stream.Token S)) (expected: List (ErrorItem (Stream.Token S)))
-| fancy (offset: Nat) (expected : List (ErrorFancy E))
+inductive ParseError (s : Stream S) where
+| trivial (offset: Nat) (unexpected: ErrorItem s.Token) (expected: List (ErrorItem s.Token))
+| fancy (offset: Nat) (expected : List ErrorFancy)
 
-structure State (S E: Type) [s: Stream S] where
+structure State (S E : Type) (s : Stream S) where
   stateInput       : S
   stateOffset      : Nat
   statePosState    : PosState S
-  stateParseErrors : List (@ParseError S E s)
+  stateParseErrors : List (ParseError s)
 
 abbrev Hints (T : Type) := List (ErrorItem T)
 
-inductive Result (S E A: Type) [s: Stream S] where
+inductive Result (S E A: Type) (s : Stream S) where
 | ok (x : A)
```

Comment:
This type is actually the same as Either. Why don't we get rid of `Result` in favour of `Either`?

Reply:
Sure